alpha/apps/kaltura/modules/extwidget/actions/downloadAction.class.php

```diff
@@ -238,27 +242,35 @@ private function dumpFile($file_path, $file_name, $limit_file_size = 0, $key = n
 			kFileUtils::dumpFile($file_path, $mime_type, null, $limit_file_size, $key, $iv, $fileSize, $allowRemote, $fileExt);
 		}
 	}
-	
-	private function handleFileSyncRedirection(FileSyncKey $syncKey)
-	{
-		list($fileSync, $local) = kFileSyncUtils::getReadyFileSyncForKey($syncKey, true, false);
-		
-		if (is_null($fileSync))
-			KExternalErrors::dieError(KExternalErrors::FILE_NOT_FOUND);
 
-		if (!$local && !in_array($fileSync->getDc(), kStorageExporter::getPeriodicStorageIds()))
+	private function handleFileSyncRedirection($fileSync, asset $flavorAsset, $entryId, $fileName, $isDir)
+	{
+		$downloadDeliveryProfile = myPartnerUtils::getDownloadDeliveryProfile($fileSync->getDc(), $entryId);
+		if($downloadDeliveryProfile && $flavorAsset)
+		{
+			$url = $this->getDownloadRedirectUrl($downloadDeliveryProfile, $flavorAsset, $fileName, $isDir);
+		}
+		else if(in_array($fileSync->getDc(), kStorageExporter::getPeriodicStorageIds()) || in_array( $fileSync->getDc(), kDataCenterMgr::getSharedStorageProfileIds()))
```

Comment:
No need to check kStorageExporter::getPeriodicStorageIds(), as both are using the ID